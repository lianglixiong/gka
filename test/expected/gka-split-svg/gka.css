.animation {
    width: 670px;
    height: 23px;
    background-image: url("./img/new1.png");
    background-repeat: no-repeat;
    animation-name: keyframes-split;
    animation-duration: 0.12s;
    animation-delay: 0s;
    animation-iteration-count: infinite;
    animation-fill-mode: forwards;
    animation-timing-function: steps(1);
}

@keyframes keyframes-split {
    0% {
        width: 670px;
        height: 23px;
        background-image: url("./img/new1.png");
    }
    33.33% {
        background-image: url("./img/new2.png");
    }
    66.67%,
    100% {
        background-image: url("./img/new3.png");
    }
}