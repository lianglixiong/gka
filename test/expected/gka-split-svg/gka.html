<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=0,maximum-scale=1">
    
    <title>gka-preview</title>
<link href="./gka.css" rel="stylesheet" type="text/css">
</head>
<body>
    <div style="position: fixed; bottom: 60px;">
        widthï¼š<input type="range" min="0" max="1340" value="670" oninput="container.style.width=this.value+'px';">
    </div>
    
    <svg id="container" viewBox="0, 0, 670, 23" style="width: 670px">
        <foreignobject width="670" height="23">
            <div id="gka" class="gka-base"></div>
        </foreignobject>
    </svg>
    <script>
    function preloadImage(names, cb, prefix){
        window.gkaCache = window.gkaCache || [];
        var n = 0,img,imgs = {};
        names.forEach(function(name) {
            img = new Image();
            window.gkaCache.push(img);
            img.onload = (function(name, img) {
                return function() {
                    imgs[name] = img;
                    (++n === names.length) && cb && cb(imgs);
                }
            })(name, img);
            img.src = (prefix || '') + name;
        });
    }
    preloadImage(["new1.png","new2.png","new3.png"], function() {
        document.getElementById('gka').className += " animation"
    }, "img/")
    </script>
</body>
</html>